kong/plugins/zipkin/handler.lua

```diff
@@ -283,40 +266,8 @@ function ZipkinLogHandler:log(conf) -- luacheck: ignore 212
     ngx_ctx.KONG_BODY_FILTER_ENDED_AT and ngx_ctx.KONG_BODY_FILTER_ENDED_AT * 1000
     or now_mu
 
-  if ngx_ctx.KONG_REWRITE_START and ngx_ctx.KONG_REWRITE_TIME then
-    -- note: rewrite is logged on the request span, not on the proxy span
-    local rewrite_finish_mu = (ngx_ctx.KONG_REWRITE_START + ngx_ctx.KONG_REWRITE_TIME) * 1000
-    zipkin.request_span:annotate("krf", rewrite_finish_mu)
-  end
-
-  if subsystem == "http" then
-    -- annotate access_start here instead of in the access phase
-    -- because the plugin access phase is skipped when dealing with
-    -- requests which are not matched by any route
-    -- but we still want to know when the access phase "started"
-    local access_start_mu =
-      ngx_ctx.KONG_ACCESS_START and ngx_ctx.KONG_ACCESS_START * 1000
-      or proxy_span.timestamp
-    proxy_span:annotate("kas", access_start_mu)
-
-    local access_finish_mu =
-      ngx_ctx.KONG_ACCESS_ENDED_AT and ngx_ctx.KONG_ACCESS_ENDED_AT * 1000
-      or proxy_finish_mu
-    proxy_span:annotate("kaf", access_finish_mu)
-
-    if not zipkin.header_filter_finished then
-      proxy_span:annotate("khf", now_mu)
-      zipkin.header_filter_finished = true
-    end
-
-    proxy_span:annotate("kbf", now_mu)
-
-  else
-    local preread_finish_mu =
-      ngx_ctx.KONG_PREREAD_ENDED_AT and ngx_ctx.KONG_PREREAD_ENDED_AT * 1000
-      or proxy_finish_mu
-    proxy_span:annotate("kpf", preread_finish_mu)
-  end
+  tag_with_durations(request_span)
+  tag_with_durations(proxy_span)
```

Comment:
There could be a config option to use tags or spans, however it would increase the complexity of the code and tests quite a bit.